### PR TITLE
Include docs on building IB/RDMA support

### PR DESCRIPTION
**Pull request by quasiben**
When https://github.com/rapidsai/ucx-split-feedstock/pull/40 is merged we should have docs on how to build UCX properly for IB/RDMA support

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -68,6 +72,10 @@ UCX
     git clone https://github.com/openucx/ucx
     cd ucx
     git checkout v1.8.x
+    # apply UCX IB registration patches
+    curl -L https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/add-page-alignment.patch > /tmp/add-page-alignment.patch
+    curl -L https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/ib_registration_cache.patch > /tmp/ib_registration_cache.patch
```
Comment (jakirkham) [inline docs/source/install.rst:77]: Could just do `curl -LO` to download them to a file of the same name in the current working directory.

```suggestion
    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/add-page-alignment.patch
    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/ib_registration_cache.patch
```

File: docs/source/install.rst [hunk 2]
```diff
@@ -68,6 +72,10 @@ UCX
...
+    git apply /tmp/ib_registration_cache.patch && git apply /tmp/add-page-alignment.patch
```
Comment (jakirkham) [inline docs/source/install.rst:78]: Could just do `curl -LO` to download them to a file of the same name in the current working directory.

```suggestion
    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/add-page-alignment.patch
    curl -LO https://raw.githubusercontent.com/rapidsai/ucx-split-feedstock/master/recipe/ib_registration_cache.patch
    git apply ib_registration_cache.patch && git apply add-page-alignment.patch
```

File: docs/source/install.rst [hunk 3]
```diff
@@ -89,3 +123,11 @@ UCX-Py
...
     # or for develop build
     pip install -v -e .
+
+
+##
+
+
+
+(ucx) bzaitlen@dgx12:/datasets/bzaitlen/GitRepos/ucx$ git apply /tmp/ib_registration_cache.patch
+(ucx) bzaitlen@dgx12:/datasets/bzaitlen/GitRepos/ucx$ git apply /tmp/add-page-alignment.patch
```
Comment (jakirkham) [inline docs/source/install.rst:133]: Is this mean to still be here or can this be dropped?

```suggestion
```

**Comment by jakirkham** [comment]
Thanks Ben! 😄 Made a couple minor suggestions above.

File: docs/source/install.rst [hunk 4]
```diff
@@ -77,6 +85,32 @@ UCX
...
+
+::
+
+    (ucx) user@dgx:~$ ofed_info -s
+    OFED-internal-4.7-3.2.9
+
+If OFED drivers are not installed on the machine, you can download drivers at directly from `Mellanox <https://www.mellanox.com/products/infiniband-drivers/linux/mlnx_ofed>`_.  For versions older than 5.1 click on, *archive versions*.
+
+
+To build UCX with IB/RDMA support, include the ``--with-rdmacm``, ``--with-verbs``, and ``--with-cm`` build flags.  For example:
```
Comment (pentschev) [inline docs/source/install.rst:101]:
```suggestion
To build UCX with IB/RDMA support, include the ``--with-rdmacm`` and ``--with-verbs`` build flags.  For example:
```

Comment (pentschev) [reply docs/source/install.rst:101]: `cm` is deprecated, according to a comment by Yossi in Slack:

> there are several things called "cm" in different contexts, in particular usage of libibcm package and --with-cm config param are considered deprecated and will probably be removed

File: docs/source/install.rst [hunk 5]
```diff
@@ -77,6 +85,32 @@ UCX
...
+
+::
+     ../contrib/configure-release --enable-mt
+    --prefix="$CONDA_PREFIX" \
+    --with-cuda="$CUDA_HOME" \
+    --enable-mt \
+    --with-cm \
```
Comment (pentschev) [inline docs/source/install.rst:108]:
```suggestion
```

File: docs/source/install.rst [hunk 6]
```diff
@@ -38,6 +38,10 @@ Source
 
 The following instructions assume you'll be using ucx-py on a CUDA enabled system and is in a `Conda environment <https://docs.conda.io/projects/conda/en/latest/>`_.
 
+.. note::
+    As of version 0.15, the UCX conda package will no longer build with IB/RDMA included.  This is largely due in part to compatibility issues
```
Comment (pentschev) [inline docs/source/install.rst:42]:
```suggestion
    As of version 0.15, the UCX conda package build will no longer include IB/RDMA support.  This is largely due in part to compatibility issues
```

File: docs/source/install.rst [hunk 7]
```diff
@@ -68,6 +72,10 @@ UCX
     git clone https://github.com/openucx/ucx
     cd ucx
     git checkout v1.8.x
+    # apply UCX IB registration patches
```
Comment (pentschev) [inline docs/source/install.rst:75]:
```suggestion
    # apply UCX IB registration cache patches, improves overall
    # IB performance when using a memory pool
```

**Review by pentschev — CHANGES_REQUESTED** [review]
Thanks @quasiben , I've added some changes, but it looks good overall. :)

**Comment by pentschev** [comment]
@quasiben I pushed https://github.com/rapidsai/ucx-py/pull/584/commits/84c0db8e75263f9db60a96630e5320cfdb876cf6 too, I noticed you accepted the other suggestion to get rid of `--with-cm` so I assumed you only missed this. Hope you don't mind. :)

**Comment by quasiben** [comment]
> @quasiben I pushed 84c0db8 too, I noticed you accepted the other suggestion to get rid of --with-cm so I assumed you only missed this. Hope you don't mind. :)

Not at all

**Review by pentschev — APPROVED** [review]
LGTM, thanks @quasiben .

**Comment by quasiben** [comment]
Thanks @pentschev and @jakirkham for the review.  I pushed one more commit to update text a bit and noticed a small error.  If you have time can you review once more and merge if you agree?

**Comment by pentschev** [comment]
> Thanks @pentschev and @jakirkham for the review. I pushed one more commit to update text a bit and noticed a small error. If you have time can you review once more and merge if you agree?

LGTM, I'm fine merging as is, will leave it up to @jakirkham once he gets a chance to take a final look.

**Comment by quasiben** [comment]
I'm fine waiting

**Comment by jakirkham** [comment]
Thanks Ben! Also thanks Peter for reviewing! 😄

**Pull request merged**